core/src/test/java/org/infinispan/persistence/WriteBehindFaultToleranceTest.java

```diff
@@ -60,11 +60,10 @@ public void testBlockingOnStoreAvailabilityChange() {
       store.setAvailable(false);
       assertFalse(store.isAvailable());
 
-      TestingUtil.sleepThread(AVAILABILITY_INTERVAL * 5);
       // PM & AsyncWriter should still be available as the async modification queue is not full
       assertTrue(asyncWriter.isAvailable());
       assertFalse(TestingUtil.extractField(asyncWriter, "delegateAvailable"));
-      assertTrue(pm.isAvailable());
+      eventually(pm::isAvailable);
```

Comment:
Doesn't the fact that we have to wait for the PM to become available contradict the comment above?

Reply:
Yes you're right. I was obviously in auto-pilot when I changed this line.